backend/src/v2/component/launcher_v2.go

```diff
@@ -427,6 +433,46 @@ func prettyPrint(jsonStr string) string {
 
 const OutputMetadataFilepath = "/tmp/kfp_outputs/output_metadata.json"
 
+// We overwrite this as a DI mechanism for testing getLogWriter. We could make
+// it an explicit input to getLogWriter, but then it would have to be an input
+// in getLogWriter's caller as well.
+var osCreateFunc = os.Create
```

Comment:
optional: another recurring pattern we have been using is just passing an `opts` argument around which feels a little more extensible/cleaner to me if we need to do something similar elsewhere, I'm not strongly opinionated on this one

Reply:
Technically, passing an options object is more correct. The only problem is that `execute`, which calls `getLogWriter`, and `executeV2`, which calls `execute`, and `Execute`, which calls `executeV2` would all require updated function signatures to handle the `opts` parameter. Many of these functions lack unit tests as well. With that in mind, I think that the global is probably an acceptable compromise for the time being. We can update the function signatures and implement proper unit tests in a separate pass.